integrations/google-analytics-4/lib/index.js

```diff
@@ -0,0 +1,310 @@
+  }
+
+  /**
+   * Privacy and Advertising Settings
+   */
+
+  // Disable All Advertising
+  // https://developers.google.com/analytics/devguides/collection/ga4/display-features#disable_all_advertising_features
+  if (opts.disableAllAdvertisingFeatures) {
+    config.push(['set', 'allow_google_signals', false]);
+  }
+
+  // Disable Advertising Personalization
+  // https://developers.google.com/analytics/devguides/collection/ga4/display-features#disable_advertising_personalization
+  if (opts.disableAdvertisingPersonalization) {
+    config.push(['set', 'allow_ad_personalization_signals', false]);
+  }
+
+  // Load gtag.js using the first measurement ID, then configure
+  // using the `config` commands built above.
+  var self = this;
+  this.load({ measurementId: measurementIds[0] }, function() {
+    for (var i = 0; i < config.length; i++) {
+      window.gtag.apply(null, config[i]);
+    }
+
+    self.ready();
+  });
+};
+
+/**
+ * Loaded?
+ *
+ * @api private
+ * @return {boolean}
+ */
+GA4.prototype.loaded = function() {
+  return !!(
+    window.ga4DataLayer && Array.prototype.push !== window.ga4DataLayer.push
+  );
+};
+
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Facade.Identify} event
+ */
+GA4.prototype.identify = function(identify) {
+  var opts = this.options;
+  var userPropertyMappings = opts.userProperties;
+
+  var userProperties = {};
+
+  // Map all customer-defined user property mappings.
+  for (var eventField in userPropertyMappings) {
+    if (!userPropertyMappings.hasOwnProperty(eventField)) {
+      continue;
+    }
+
+    var userProp = userPropertyMappings[eventField];
+    var value = identify.proxy(eventField);
+
+    userProperties[userProp] = value;
+  }
+
+    /**
+     * Map the user_id property the Send User ID setting is enabled. Note that the user ID
+     * can be appended as part of the user_properties object instead of being configured by
+     * an explicit command.
+     * https://developers.google.com/analytics/devguides/collection/ga4/cookies-user-id#set_user_id
+     */
+  var userId = identify.userId();
+  if (opts.sendUserId && userId) {
+    userProperties.user_id = userId;
+  }
+
+  if (Object.keys(userProperties).length) {
+    window.gtag('set', 'user_properties', userProperties);
+  }
+};
+
+/**
+ * Group
+ *
+ * @api public
+ * @param {Facade.Group} group
+ */
+GA4.prototype.group = function(group) {
+  var props = reject({
+    group_id: group.groupId()
+  });
+
+  window.gtag('event', 'join_group', props);
+};
+
+/**
+ * Page
+ *
+ * @api public
+ * @param {Facade.Page} page
+ */
+GA4.prototype.page = function(page) {
+  // If the Disable Page View Measurement setting is set to false then
+  // don't handle page calls to avoid duplicate page_view events.
+  if (!this.options.disablePageViewMeasurement) {
+    return;
+  }
+
+  // Track categorized pages setting needed here?
+
+  var props = page.properties();
+  var name = page.fullName();
+
+  // var language = ;
```

Comment:
Not sure if we have or are working on a plan for language? I see in the SDD that Segment doesn't have it specced for a.js events.

Reply:
I don't think it is part of the Segment spec which is why I left it commented. It is mapped in the SDD but I don't think it's crucial for the pilot version. I can remove for now and re-add if there are asks for it!